src/charm.py

```diff
@@ -246,9 +244,11 @@ def __init__(self, *args):
             relation_name="logging",
         )
 
-        postgresql_db_port = ServicePort(5432, name="database")
-        patroni_api_port = ServicePort(8008, name="api")
-        self.service_patcher = KubernetesServicePatch(self, [postgresql_db_port, patroni_api_port])
+        if JujuVersion.from_environ().supports_open_port_on_k8s:
+            try:
+                self.unit.set_ports(5432, 8008)
+            except ModelError:
+                logger.exception("failed to open port")
```

Comment:
Copied from the Mysql charm: https://github.com/canonical/mysql-k8s-operator/blob/1301de05c224e0557b373445ff333fa85d404710/src/charm.py#L600-L604